docs/runtimes/create-and-use-functions.ipynb

```diff
@@ -0,0 +1,550 @@
+    "    name=\"notebook\", func=\"nb.ipynb\", kind=\"serving\", \n",
+    "    image=\"mlrun/ml-models\", requirements=[\"pandas==1.3.5\"]\n",
+    ")\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6b36faf7",
+   "metadata": {},
+   "source": [
+    "You can also create an MLRun function out of the current Jupyter notebook you are running in. To do this, simply ommit the `func` parameter in `set_function`. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0204fd5d",
+   "metadata": {},
+   "source": [
+    "<a id=\"multiple-source-files\"></a>\n",
+    "### Multiple Source Files"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4f32f1ea",
+   "metadata": {},
+   "source": [
+    "If your code requires additional files or external libraries, you will need to use a source that supports multiple files such as Git, an archive (zip/tar/etc.), or V3IO file share. This approach (especially using a Git repo) pairs well with MLRun Projects.\n",
+    "\n",
+    "To do this, you must:\n",
+    "- Provide `with_repo=True` when creating your function via `project.set_function(...)`\n",
+    "- Set project source via `project.set_source(source=...)`\n",
+    "\n",
+    "This instructs MLRun to load source code from the git repo/archive/file share associated with the project. There are two ways to load these additional files:\n",
+    "\n",
+    "#### Load Code from Container\n",
+    "The function is built once. *This is the preferred approach for production workloads*:\n",
+    "```python\n",
+    "project.set_source(source=\"git://github.com/mlrun/project-archive.git\")\n",
+    "\n",
+    "fn = project.set_function(\n",
+    "    name=\"myjob\", handler=\"job_func.job_handler\",\n",
+    "    image=\"mlrun/mlrun\", kind=\"job\", with_repo=True,\n",
+    ")\n",
+    "\n",
+    "project.build_function(fn)\n",
+    "```\n",
+    "\n",
+    "#### Load Code at Runtime\n",
+    "The function pulls the source code at runtime. *This is a simpler approach during development that allows for making code changes without re-building the image each time*:\n",
+    "\n",
+    "```python\n",
+    "archive_url = \"https://s3.us-east-1.wasabisys.com/iguazio/project-archive/project-archive.zip\"\n",
+    "project.set_source(source=archive_url, pull_at_runtime=True)\n",
+    "\n",
+    "fn = project.set_function(\n",
+    "    name=\"nuclio\", handler=\"nuclio_func:nuclio_handler\",\n",
+    "    image=\"mlrun/mlrun\", kind=\"nuclio\", with_repo=True,\n",
+    ")\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "969cc92e",
+   "metadata": {},
+   "source": [
+    "<a id=\"import-existing-function\"></a>\n",
+    "### Import or Use Existing Function"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0c5adbe0",
+   "metadata": {},
+   "source": [
+    "If you already have an MLRun function that you would like to import, you can do so from multiple locations such as YAML, function marketplace, and MLRun DB."
```

Comment:
"If you already have an MLRun function that you want to import, you can do so from multiple locations such as YAML, function marketplace, and MLRun DB."